{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"자연어처리 6-2수정.ipynb","provenance":[],"collapsed_sections":[],"authorship_tag":"ABX9TyNrNYGaBuwF78EIi4C8XNlP"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":722},"id":"6sc0K0wOMAXF","executionInfo":{"status":"ok","timestamp":1619431848475,"user_tz":-540,"elapsed":4334,"user":{"displayName":"이덕수","photoUrl":"","userId":"04402217723891097609"}},"outputId":"4b2c2b22-e5a8-43d6-d779-7709ee218bac"},"source":["!pip install spacy==2.2.2\n","import spacy\n","\n","nlp = spacy.load('en_core_web_sm')\n","\n","doc = nlp('The huge pizza no eat anyone but monster eat is possible do you agree?')\n","\n","for token in doc:\n","  print(token.text, token.dep_, token.head.text)\n","\n","  from spacy import displacy\n","\n","displacy.render(doc, style='ent', jupyter=True)"],"execution_count":2,"outputs":[{"output_type":"stream","text":["Requirement already satisfied: spacy==2.2.2 in /usr/local/lib/python3.7/dist-packages (2.2.2)\n","Requirement already satisfied: setuptools in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (56.0.0)\n","Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (1.0.5)\n","Requirement already satisfied: requests<3.0.0,>=2.13.0 in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (2.23.0)\n","Requirement already satisfied: wasabi<1.1.0,>=0.3.0 in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (0.8.2)\n","Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (3.0.5)\n","Requirement already satisfied: blis<0.5.0,>=0.4.0 in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (0.4.1)\n","Requirement already satisfied: srsly<1.1.0,>=0.1.0 in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (1.0.5)\n","Requirement already satisfied: numpy>=1.15.0 in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (1.19.5)\n","Requirement already satisfied: importlib-metadata>=0.20; python_version < \"3.8\" in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (3.10.1)\n","Requirement already satisfied: thinc<7.4.0,>=7.3.0 in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (7.3.1)\n","Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (2.0.5)\n","Requirement already satisfied: plac<1.2.0,>=0.9.6 in /usr/local/lib/python3.7/dist-packages (from spacy==2.2.2) (1.1.3)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/dist-packages (from requests<3.0.0,>=2.13.0->spacy==2.2.2) (1.24.3)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests<3.0.0,>=2.13.0->spacy==2.2.2) (2020.12.5)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests<3.0.0,>=2.13.0->spacy==2.2.2) (2.10)\n","Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests<3.0.0,>=2.13.0->spacy==2.2.2) (3.0.4)\n","Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/dist-packages (from importlib-metadata>=0.20; python_version < \"3.8\"->spacy==2.2.2) (3.4.1)\n","Requirement already satisfied: typing-extensions>=3.6.4; python_version < \"3.8\" in /usr/local/lib/python3.7/dist-packages (from importlib-metadata>=0.20; python_version < \"3.8\"->spacy==2.2.2) (3.7.4.3)\n","Requirement already satisfied: tqdm<5.0.0,>=4.10.0 in /usr/local/lib/python3.7/dist-packages (from thinc<7.4.0,>=7.3.0->spacy==2.2.2) (4.41.1)\n","The det pizza\n","huge amod pizza\n","pizza nsubj eat\n","no neg eat\n","eat ROOT eat\n","anyone dobj eat\n","but cc anyone\n","monster compound eat\n","eat conj anyone\n","is ROOT is\n","possible acomp is\n","do aux agree\n","you nsubj agree\n","agree ROOT agree\n","? punct agree\n"],"name":"stdout"},{"output_type":"stream","text":["/usr/lib/python3.7/runpy.py:193: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.\n","  \"__main__\", mod_spec)\n"],"name":"stderr"},{"output_type":"display_data","data":{"text/html":["<div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The huge pizza no eat anyone but monster eat is possible do you agree?</div>"],"text/plain":["<IPython.core.display.HTML object>"]},"metadata":{"tags":[]}}]}]}